packages/bee-q/src/components/toast/bq-toast.tsx

```diff
@@ -0,0 +1,135 @@
+import { h, Component, Element, Prop, Watch, Method, State, Host } from '@stencil/core';
+import { getColorCSSVariable, validatePropValue } from '../../shared/utils';
+import { TOAST_TYPE, TToastType } from './bq-toast.types';
+
+/**
+ * @part base - The component's internal wrapper of the Toast component.
+ * @part icon - `<div>` container element of toast icon component.
+ * @part text - `<div>` container element of toast text slot.
+ */
+@Component({
+  tag: 'bq-toast',
+  styleUrl: './scss/bq-toast.scss',
+  shadow: true,
+})
+export class BqToast {
+  // Own Properties
+  // ====================
+
+  // Reference to host HTML element
+  // ===================================
+  @Element() el!: HTMLBqToastElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+  /** State of Toast */
+  @State() private shouldShowToast = false;
+
+  // Public Property API
+  // ========================
+  /** Type of Toast */
+  @Prop({ reflect: true, mutable: true }) type: TToastType = 'default';
+  /** Text color of Toast */
+  @Prop({ reflect: true }) textColor: string;
+  /** Should show icon of Toast */
+  @Prop({ reflect: true }) showIcon = false;
+  /** Should hide Toast after period of time */
+  @Prop({ reflect: true }) autoCloseTime: number;
+
+  // Prop lifecycle events
+  // =======================
+  @Watch('type')
+  checkPropValues() {
+    validatePropValue(TOAST_TYPE, 'default', this.el, 'type');
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+  componentWillLoad() {
+    this.checkPropValues();
+  }
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  /** Trigers function to show toast */
+  @Method()
+  async showToast() {
+    this.shouldShowToast = true;
+    const timeout = !this.autoCloseTime ? 5000 : this.autoCloseTime;
+    setTimeout(() => {
+      this.shouldShowToast = false;
+    }, timeout);
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+  private getColorAndIcon = () => {
+    const type = this.type;
+    const textColor = this.textColor;
+    const defaultColors = {
```

Comment:
It can be split into two functions
  private getColorAndIcon = () => {
```
private get iconColor(){}

private get icon(){}
```

this way you can just call it directly as property even though is computed `this.iconColor `